cwl/src/main/scala/cwl/model.scala

```diff
@@ -47,7 +47,7 @@ case class WorkflowStepInput(
       //lookup each of our source Ids, failing if any of them are missing
       inputTypes <- sources.traverse[ErrorOr, WomType](lookupId).toEither
       //we may have several sources, we make sure to have a type common to all of them
-      inputType = WomType.lowestCommonSubtype(inputTypes)
+      inputType = WomType.homogeneousTypeFromTypes(inputTypes)
```

Comment:
Do we actually want to constraint the source to being of the same type ? Or does the spec allow for sources of different types ?

Reply:
This falls under MultipleInputFeatureRequirement, which I figured I would tackle next.

That being said, the spec doesn't address the case of input sources being different type, so I'm really just giving a best guess here. I can comment to that effect.

Reply:
Oh right my bad, for some reason I thought this PR was for `MultipleInputFeatureRequirement`, I didn't realize there was a Step input requirement too
I would be surprised if inputs could only be of the same type but :man_shrugging: